app/src/userpages/components/TOCPage/index.jsx

```diff
@@ -0,0 +1,60 @@
+// @flow
+
+import React, { Fragment, Component, type Node } from 'react'
+import { Col, Row } from 'reactstrap'
+
+import TOCSection from './TOCSection'
+import styles from './tocPage.pcss'
+
+type Props = {
+    title?: string,
+    children: ?Node,
+}
+
+class TOCPage extends Component<Props> {
+    static Section = TOCSection
+
+    parseMenu = () => React.Children.map(this.props.children, (child) => {
+        if (child.type === TOCSection) {
```

Comment:
It's a bit of defensive coding, I suppose we will never abuse it :) But I'll leave it in because the titles are picked from the props of only the `TOCSection` component. You could potentially put something else there but it won't show in the TOC.